draft-irtf-cfrg-opaque.md

```diff
@@ -403,26 +398,32 @@ used in the AKE.
 Registration is executed between a client C and a
 server S. It is assumed S can identify C and the client can
 authenticate S during this registration phase. This is the only part
-in OPAQUE that requires an authenticated channel, either physical, out-of-band,
+in OPAQUE that requires an authenticated and confidential channel, either physical, out-of-band,
 PKI-based, etc. This section describes the registration flow, message encoding,
 and helper functions. Moreover, C has a key pair (client_private_key, client_public_key) for an AKE protocol
 which is suitable for use with OPAQUE; See {{online-phase}}. The private-public keys (client_private_key, client_public_key) may be randomly generated (using a cryptographically secure pseudorandom number generator) for the account or provided by the calling client.
 Clients MUST NOT use the same key pair (client_private_key, client_public_key) for two different accounts.
 
-To begin, C chooses its password, and S chooses its own pair of private-public
-AKE keys (server_private_key, server_public_key) for use with the AKE. S can use
-the same pair of keys with multiple clients. These steps can happen offline, i.e.,
-before the registration phase. Once complete, the registration process proceeds as follows:
+## Setup Phase {#setup-phase}
+
+In a setup phase, C chooses its password, and S chooses its own pair of private-public
+AKE keys (server_private_key, server_public_key) for use with the AKE, along with a Nh-byte oprf_seed. S can use
+the same pair of keys with multiple clients, and can opt to use multiple seeds (so long as they are
+kept consistent for each client). These steps can happen offline, i.e., before the registration phase.
+
+Once complete, the registration process proceeds as follows.
+
+## Credential Registration
 
 ~~~
- Client (password, creds)            Server (server_private_key, server_public_key)
+ Client (password, creds)            Server (server_private_key, server_public_key, oprf_seed)
  --------------------------------------------------------------------
  (request, blind) = CreateRegistrationRequest(password)
 
                                request
                       ------------------------->
 
-            (response, oprf_key) = CreateRegistrationResponse(request, server_public_key)
+            (response, oprf_key) = CreateRegistrationResponse(request, server_public_key, credential_identifier, oprf_seed)
```

Comment:
Where does `credential_identifier` come from? It should probably be included in the Server inputs on L419.

Reply:
Added!